shell/edit/workload/index.vue

```diff
@@ -833,57 +891,58 @@ export default {
       :done-route="doneRoute"
       :subtypes="workloadSubTypes"
       :apply-hooks="applyHooks"
+      :value="value"
       @finish="save"
       @select-type="selectType"
       @error="e=>errors = e"
     >
-      <div class="row">
-        <div class="col span-12">
-          <NameNsDescription :value="value" :extra-columns="nameNsColumns" :mode="mode" @change="name=value.metadata.name">
-            <template #schedule>
-              <LabeledInput
-                v-model="spec.schedule"
-                type="cron"
-                required
-                :mode="mode"
-                :label="t('workload.cronSchedule')"
-                placeholder="0 * * * *"
-              />
-            </template>
-            <template #replicas>
-              <LabeledInput
-                v-model.number="spec.replicas"
-                type="number"
-                min="0"
-                required
-                :mode="mode"
-                :label="t('workload.replicas')"
-              />
-            </template>
-            <template #service>
-              <LabeledSelect
-                v-model="spec.serviceName"
-                option-label="metadata.name"
-                :reduce="service=>service.metadata.name"
-                :mode="mode"
-                :label="t('workload.serviceName')"
-                :options="headlessServices"
-                required
-              />
-            </template>
-          </NameNsDescription>
+      <NameNsDescription
+        :value="value"
+        :mode="mode"
+        @change="name=value.metadata.name"
+      />
+      <div class="row mb-10">
+        <div v-if="isCronJob" class="col span-4">
```

Comment:
Instead of adding extra columns into NameNsDescription through slots, the workload form now just adds the extra columns onto the second row in the form using v-if statements to render the ones appropriate for each workload type.